servicetest/testframework/lib.py

```diff
@@ -32,9 +32,105 @@
 import dbus.mainloop.glib
+import os.path
 from gi.repository import GObject
 
 
+class ConfigFile(object):
+    """ Represents the configuration file for SoftwareContainer
+        All keys and values have to be strings.
+
+        The format of a config file is as follows:
+
+        [Group1]
+        item1 = value1
+        item2 = value2
+
+        [Group2]
+        item1 = value1
+        item2 = value2
+
+        Initialization is done as follows:
+
+        config = ConfigFile("/where/to/save/me", {
+            "Group1": {
+                "item1": "value1",
+                "item2": "value2"
+            },
+
+            "Group2": {
+                "item1": "value1",
+                "item2": "value2"
+            }
+        })
+    """
+
+    def __init__(self, location, full_config):
+        """ Create a config object from a given dict of dicts.
+        """
+        self.__filename = location
+        self.__groups = full_config
+
+    def filename(self):
+        return self.__filename
+
+    def config_as_string(self):
+        """ Convert this object into a string that can be written to a config file
+        """
+
+        # Copy the values from the defaults to the given config
+        # Only add them if there is no matching value already in the given config.
+        defaults = self.__default_config()
+        for group in defaults:
+            if group not in self.__groups:
+                self.__groups[group] = defaults[group]
+            else:
+                for item in defaults[group]:
+                    if item not in self.__groups[group]:
+                        self.__groups[group][item] = defaults[group][item]
+
+        result = ""
+        for group in self.__groups:
+            result += "[" + group + "]\n"
+            for item in self.__groups[group]:
+                result += item + " = " + self.__groups[group][item] + "\n"
+
+        return result
+
+    def __default_config(self):
+        """ Read the softwarecontainer-config prepared by cmake, and return its values as a digt.
+            This expects the file to be available in the same directory as this file.
+
+            The file, and the dict, will only contain values that are mandatory options, and as
+            such has no default values within SoftwareContainer. These are the values that one
+            has to have in a config file for the agent to start at all. Any other config values
+            have reasonable defaults and are therefore commented out in the config file.
+        """
+        conf = dict()
+        group = None
+        config_path = os.path.join(os.path.dirname(__file__), "softwarecontainer-config")
+        with open(config_path, "r") as fh:
+            while True:
+                line = fh.readline()
+                if not line:
+                    return conf
+                else:
+                    line = line.strip()
+                    if len(line) == 0 or line[0] == '\n' or line[0] == '#':
```

Comment:
Maybe add a short comment here what this means, i.e. what are we ignoring? Empty lines and newlines-only are easy to guess why, but it's not obvious that the '#' is related to the top comment about what's extracted or not.

Like: "Skip empty lines and comments in the file"